src/services/updateZendeskTicket.spec.ts

```diff
@@ -0,0 +1,134 @@
+// Under test
+import { updateZendeskTicket } from './updateZendeskTicket'
+// Dependencies
+import { makeHttpsRequest, base64Encode } from './httpsRequestUtils'
+import { retrieveZendeskApiSecrets } from './retrieveZendeskApiSecrets'
+import { ZendeskApiSecrets } from '../types/zendeskApiSecrets'
+import { exampleEventBody } from '../testUtils/events/exampleEventBody'
+
+// Mocking retrieveZendeskApiSecrets
+jest.mock('./retrieveZendeskApiSecrets', () => ({
+  retrieveZendeskApiSecrets: jest.fn()
+}))
+const mockRetrieveZendeskApiSecrets = retrieveZendeskApiSecrets as jest.Mock<
+  Promise<ZendeskApiSecrets>
+>
+const givenSecretKeysSet = (secrets: ZendeskApiSecrets) => {
+  mockRetrieveZendeskApiSecrets.mockResolvedValue(secrets)
+}
+const allSecretKeys: ZendeskApiSecrets = {
+  zendeskApiKey: 'myZendeskApiKey',
+  zendeskApiUserId: 'myZendeskApiUserId',
+  zendeskApiUserEmail: 'myZendeskApiUserEmail'
+}
+const givenAllSecretsAvailable = () => {
+  givenSecretKeysSet(allSecretKeys)
+}
+
+// Mocking base64Encode and makeHttpsRequest
+jest.mock('./httpsRequestUtils', () => ({
+  base64Encode: jest.fn(),
+  makeHttpsRequest: jest.fn()
+}))
+const mockBase64Encode = base64Encode as jest.Mock<string>
+const generateTestValidAuth = () => {
+  mockBase64Encode.mockReturnValue('VALIDEncodedAuthValue')
+}
+const generateTestInvalidAuth = () => {
+  mockBase64Encode.mockReturnValue('INVALIDEncodedAuthValue')
+}
+const mockMakeHttpsRequest = makeHttpsRequest as unknown as jest.Mock<void>
+const givenSuccessfulApiCall = () => {
+  mockMakeHttpsRequest.mockImplementation(() => {
+    return { theReturnData: '123' }
+  })
+}
+const givenUnsuccessfulApiCall = () => {
+  mockMakeHttpsRequest.mockImplementation(() => {
+    throw new Error('There was an error.')
+  })
+}
+
+describe('updating a zendesk ticket', () => {
+  beforeEach(() => {
+    givenAllSecretsAvailable()
+  })
+  afterEach(() => {
+    jest.clearAllMocks()
+  })
+
+  it('a single api call was made with valid authentication', async () => {
+    generateTestValidAuth()
+    givenSuccessfulApiCall()
+    jest.spyOn(global.console, 'log')
```

Comment:
Would it make sense to move this into your beforeEach hook?

Reply:
done!